File: 백준알고리즘/DFS(깊이 우선 탐색)/python/연결 요소의 개수.py
import sys
sys.setrecursionlimit(10**6)
input = sys.stdin.readline

N, M = map(int, input().split())
graph = [[] for _ in range(N + 1)]
cnt = 0
visited = []

for _ in range(M):
    m1, m2 = map(int, input().split())

    graph[m1].append(m2)
    graph[m2].append(m1)


def dfs(x):
    visited.append(x)

    for i in graph[x]:
        if i not in visited:
            dfs(i)


for i in range(1, N + 1):
    if i not in visited:
        dfs(i)
        cnt += 1

print(cnt)

File: 백준알고리즘/DFS(깊이 우선 탐색)/javascript/연결 요소의 개수.js
const fs = require('fs');
const stdin = fs.readFileSync('/dev/stdin').toString().split('\n');
const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const [N, M] = input()
  .split(' ')
  .map(n => +n);
const graph = Array.from(Array(N + 1), () => Array());
const visited = Array.from(Array(N + 1), () => 0);

for (let i = 0; i < M; i++) {
  const [u, v] = input()
    .split(' ')
    .map(n => +n);

  graph[u].push(v);
  graph[v].push(u);
}

function dfs(node) {
  visited[node] = 1;

  for (const item of graph[node]) {
    if (visited[item] === 0) dfs(item);
  }
}

let cnt = 0;
for (let i = 1; i <= N; i++) {
  if (visited[i] === 0) {
    cnt++;
    dfs(i);
  }
}

console.log(cnt);
